Sources/CardPayments/Models/ServiceCard.swift

```diff
@@ -0,0 +1,79 @@
+import Foundation
+
+/// ServiceCard is used for formatting contents of `CardRequest` for `ConfirmPaymentSourceRequest`
+struct ServiceCard: Encodable {
```

Comment:
Is the naming of this supposed to be similar to a factory class on Android? I think generally we would use `Card` on iOS but curious to hear @scannillo thoughts as naming of these classes/structs is not something we've discussed much on iOS from an architecture point of view.

Reply:
I have `Card` as merchant facing class that just has fields that the customer enters. Originally this class had `Attributes` which I found inconvenient because of having it reflect exactly the format of API call. `ServiceCard` exists simply for the flexibility of formatting it and incorporating contents of the `CardResult` which now includes the `vault` object.

Reply:
I am looking at original Android refactor PR for refactor for parity with Android, having separate classes for individual APIRequests. I thought that would be too long for this PR which was started a while ago.I think the plan was for that refactor to go in before next phase of vaulting.

Reply:
2 cents from Android:

- The PayPal API has a lot of nested JSON structures that don't map nicely to `Codable`
- We had a noticeable amount of class bloat associated with making requests to the sample merchant server
-  Working with dictionaries and converting to JSON may be a better approach to give us more flexibility in the Demo app
- We do end up duplicating a lot of types to make a simple JSON request. Type safety is nice overall, but in this case it has felt like unnecessary boilerplate.

Reply:
In Swift, I've seen use of structs or class rather than raw dictionaries. I think it seems safer, less error prone and easier to read.

Reply:
Yeah I agree on the less error prone part no doubt. I guess JavaScript has the advantage here since JSON is basically `Codable` for JS.

Reply:
spicy take: I like `Codable` for the SDK but not as much for the Demo app 😅